cpp/NamingConventions.md

```diff
@@ -0,0 +1,89 @@
+# C++ Naming Conventions
+
+This document provides a set of recommendations regarding how things should be named
+in C++ projects.
+
+Because there is no automatic way to ensure that a code change follows this guideline,
+it is the responsibility of the code reviewers to enforce these rules over time.
+
+Should a project decide to not apply some or all of the recommendations
+below, it may provide written guidelines to the contributors with the rules
+that should be followed instead, in a `CONTRIBUTING.md` document for instance.
+
+## File Names
+
+Follow [File Names](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules)
+convention of Google C++ Style Guide but use the following extension instead:
+
+* C++ files: `.cpp`
+* C++ header: `.hpp`
+* C++ template implementations: `.ipp`
+
+## Use descriptive variable names
+
+```cpp
+// no
+int bufsz;
+
+// yes
+int buffer_size;
```

Comment:
one thing I come across every once in a while.. what do you do when the variable name would be really very long if you'd write it out without abbreviating?

Reply:
This is not a strict rule banning abbreviation. For instance, in the URL mentioned below:

   ```cpp
   int lstm_size;             // "LSTM" is a common machine learning abbreviation.
   ```

The idea is to be aware of the consequences of losing meaning just to save a few characters.

But I think the `buffer_size` example doesn't entirely highlight the meaning of this section. The point here is mainly to banish variables named like `count` for instance. I will make some updates. Thank you

Reply:
Hello @ohm314 
I tried to address your question in 5f3377b2
Please, _Resolve conversation_ or comment.

Reply:
Sorry @tristan0x I completely missed that. Your comment makes sense and I agree